mail/common/src/test/java/com/fsck/k9/helper/MailToTest.java

```diff
@@ -187,4 +188,52 @@ public void testCaseInsensitiveParamWrapper_withoutQueryParameters() {
 
         assertThat(Collections.<String>emptyList(), is(result));
     }
-}
+
+    @Test
+    public void testGetInReplyTo_singleMessageId() {
+        Uri uri = Uri.parse("mailto:?in-reply-to=<7C72B202-73F3@somewhere>");
+
+        MailTo mailToHelper = MailTo.parse(uri);
+
+        assertEquals(mailToHelper.getInReplyTo(), "<7C72B202-73F3@somewhere>");
+    }
+
+    @Test
+    public void testGetInReplyTo_multipleMessageIds() {
+        Uri uri = Uri.parse("mailto:?in-reply-to=<7C72B202-73F3@somewhere>;<8A39-1A87CB40C114@somewhereelse>");
```

Comment:
Message identifiers are not separated by semicolons, which is why `MessageIdParser` doesn't like this. If you remove the semicolon or replace it with a space (`%20` when encoded) your code will properly extract the first message identifier.